### PR TITLE
cleanup calendarservice mess

### PR DESCRIPTION
**Pull request by georgehrke**
- cleans up calendar service and creates calendar factory
- fixes https://github.com/nextcloud/calendar/issues/20

### CODE REVIEW

**Comment by tcitworld** [comment]
Well this is definitely cleaner. :-)

**Comment by georgehrke** [comment]
@tcitworld @raghunayyar I'd like to get this in, though there are few a still tests missing.
What do you think?
Can you please review it :)

File: js/app/factory/calendarFactory.js [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+		return canWrite;
+	};
+
+	context.color = function(props) {
+		const colorProp = props['{' + DavClient.NS_APPLE + '}calendar-color'];
+		const fallbackColor = angular.element('#fullcalendar').attr('data-defaultColor');
+
+		if (angular.isString(colorProp) && colorProp.length > 0) {
+			//some stupid clients store an alpha value in the rgb hash (like #rrggbbaa) *cough cough* Apple Calendar *cough cough*
+			//but some browsers can't parse that *cough cough* Safari *cough cough*
```
Comment (tcitworld) [inline js/app/factory/calendarFactory.js]: To me, seems like Safari supports rgba(). Maybe time to ditch that ?

File: js/app/factory/calendarFactory.js [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+	};
+
+	context.publishedAndPublishURL = function(props, publicMode) {
+		let published = false;
+		let publishurl = null;
+		let publicurl = null;
+
+		if (angular.isDefined(props['{' + DavClient.NS_CALENDARSERVER + '}publish-url'])) {
+			published = true;
+			//TODO - compare this with master
```
Comment (tcitworld) [inline js/app/factory/calendarFactory.js]: LGTM

File: js/app/factory/calendarFactory.js [hunk 3]
```diff
@@ -0,0 +1,326 @@
...
+		const currentUser = context.getUserFromUserPrincipal(userPrincipal);
+
+		const simple = context.calendarSkeleton(props, userPrincipal, publicMode);
+		simple.href = context.webcal(props);
+
+		// WebCal is obviously not writable
+		simple.writable = false;
+		simple.writableProperties = (currentUser === simple.owner);
+
+		// WebCal subscriptions are neither publishable nor shareable
```
Comment (tcitworld) [inline js/app/factory/calendarFactory.js]: Seems they are right now though. :/

Comment (georgehrke) [inline js/app/factory/calendarFactory.js]: Yes, the GUI shows the button, but it won't work.
This is a bug (in master) in the fallback, when no sharing-modes node is available.

Comment (georgehrke) [inline js/app/factory/calendarFactory.js]: @tcitworld I don't see a sharing icon next to a webcal subscription on this branch. Can you confirm this?

Comment (tcitworld) [inline js/app/factory/calendarFactory.js]: Nope, this branch is good.

Comment (georgehrke) [inline js/app/factory/calendarFactory.js]: okay, so let's get this in?

Comment (tcitworld) [inline js/app/factory/calendarFactory.js]: Please just clean the TODO above and look what's about the rgba safari workaround.

**Comment by tcitworld** [comment]
Not related, but Scrutinizer is missing the `'esversion: 6'` parameter and displays a ridiculous number of issues here.

**Comment by georgehrke** [comment]
> Not related, but Scrutinizer is missing the `'esversion: 6'` parameter and displays a ridiculous number of issues here.

I know. I tried it already but it didn't work :(

File: js/app/factory/calendarFactory.js (hunk 3 above)
Comment (georgehrke) [inline js/app/factory/calendarFactory.js]: done

**Comment by tcitworld** [comment]
:+1:

**Pull request merged**